### PR TITLE
expand docs for FixedSizeListArray

### PR DESCRIPTION
**Pull request by smiklos**
# Which issue does this PR close?

Relates to #4607

# Rationale for this change

Added extra docs to FixedSizeListArray to document its logical and physical layout.  

# What changes are included in this PR?

Just docs

# Are there any user-facing changes?

Noe

Noe

### CODE REVIEW

File: arrow-array/src/array/fixed_size_list_array.rs [hunk 1]
```diff
@@ -26,7 +26,60 @@ use arrow_schema::{ArrowError, DataType, FieldRef};
...
+///                                 ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
+///                                                         ┌ ─ ─ ─ ─ ─ ─ ─ ─┐    
+///  ┌─────────────┐                │     ┌───┐               ┌───┐ ┌──────┐      │
+///  │   [A,B]     │                      │ 1 │             │ │ 1 │ │  A   │ │ 0  
+///  ├─────────────┤                │     ├───┤               ├───┤ ├──────┤      │
+///  │    NULL     │                      │ 0 │             │ │ 1 │ │  B   │ │ 1  
+///  ├─────────────┤                │     ├───┤               ├───┤ ├──────┤      │
+///  │  [C,NULL]   │                      │ 1 │             │ │ 0 │ │ NULL │ │ 2  
+///  └─────────────┘                │     └───┘               ├───┤ ├──────┤      │
+///                                                         | │ 0 │ │ NULL │ │ 3
```
Comment (alamb) [inline arrow-array/src/array/fixed_size_list_array.rs:72]: When an array element is null, the values can be anything (valid). So I suggest using `????` to show that the contents could be something unknown:

```suggestion
///  │  [C,NULL]   │                      │ 1 │             │ │ 0 │ │ ???? │ │ 2  
///  └─────────────┘                │     └───┘               ├───┤ ├──────┤      │
///                                                         | │ 0 │ │ ???? │ │ 3  
```

File: arrow-array/src/array/fixed_size_list_array.rs [hunk 2]
```diff
@@ -26,7 +26,60 @@ use arrow_schema::{ArrowError, DataType, FieldRef};
...
+///  ├─────────────┤                │     ├───┤               ├───┤ ├──────┤      │
+///  │    NULL     │                      │ 0 │             │ │ 1 │ │  B   │ │ 1  
+///  ├─────────────┤                │     ├───┤               ├───┤ ├──────┤      │
+///  │  [C,NULL]   │                      │ 1 │             │ │ 0 │ │ NULL │ │ 2  
+///  └─────────────┘                │     └───┘               ├───┤ ├──────┤      │
+///                                                         | │ 0 │ │ NULL │ │ 3  
+///  Logical Values                 │   Validity              ├───┤ ├──────┤      │
+///                                     (nulls)             │ │ 1 │ │  C   │ │ 4  
+///                                 │                         ├───┤ ├──────┤      │
+///                                                         │ │ 0 │ │ NULL │ │ 5
```
Comment (alamb) [inline arrow-array/src/array/fixed_size_list_array.rs:76]: Same thing here:

```suggestion
///                                                         │ │ 0 │ │ ??? │ │ 5  
```

File: arrow-array/src/array/fixed_size_list_array.rs [hunk 3]
```diff
@@ -60,6 +113,9 @@ use std::sync::Arc;
 /// assert_eq!( &[3, 4, 5], list1.as_any().downcast_ref::<Int32Array>().unwrap().values());
 /// assert_eq!( &[6, 7, 8], list2.as_any().downcast_ref::<Int32Array>().unwrap().values());
 /// ```
+///
+/// [`StringArray`]: crate::array::StringArray
+/// An array of [fixed size arrays](https://arrow.apache.org/docs/format/Columnar.html#fixed-size-list-layout)
```
Comment (alamb) [inline arrow-array/src/array/fixed_size_list_array.rs:118]: I think this link formatting is a little off

```suggestion
/// [fixed size arrays]: https://arrow.apache.org/docs/format/Columnar.html#fixed-size-list-layout
```

**Comment by smiklos** [comment]
Pushed the recommended changes @alamb . Thanks for the review!

**Comment by tustvold** [comment]
Thank you

**Pull request merged**